src/cmap/connect.ts

```diff
@@ -183,7 +183,7 @@ export interface HandshakeDocument extends Document {
   ismaster?: boolean;
   hello?: boolean;
   helloOk?: boolean;
-  client: ClientMetadata;
+  client: Document;
```

Comment:
Our changes make the `client.env.name` property optional (in the case that env exists), this would be a breaking change to the `public ClientMetadata` type, so it's just easier to change it to document type.

Reply:
wow, I guess that was an oversight.  `ClientMetadata` probably should be `internal`

thanks for the context